int32 nBonus
pos[] bonusPos
